ishlib/matcher/alignment/semi_global_aln/striped.mojo

```diff
@@ -410,42 +394,44 @@ fn semi_global_aln[
             v_h = pv_h_load[j]
 
         # Lazy_F loop, disallows adjacent insertion and then deletion from SWPS3
-        # Possible speedup - check if v_f has any updates to start with
-        var break_out = False
-        for _k in range(0, width):
-            var tmp = (ZERO - gap_open_penalty).cast[
-                DType.int32
-            ]() if free_target_start_gaps else (
-                boundary[i + 1] - gap_open_penalty
-            ).cast[
-                DType.int32
-            ]()
-            var tmp2 = MIN if tmp < Int32(MIN) else tmp.cast[dt]()
-
-            v_f = v_f.shift_right[1]()
-            v_f[0] = tmp2
-
-            for j in range(0, segment_length):
-                v_h = pv_h_store[j]
-                v_h = max(v_h, v_f)
-
-                pv_h_store[j] = v_h
-
-                @parameter
-                if do_saturation_check:
-                    v_saturation_check_max = max(
-                        v_saturation_check_max, v_h - v_neg_limit
-                    )
-
-                v_h = saturating_sub(v_h, v_gap_open)
-                v_f = saturating_sub(v_f, v_gap_ext)
-
-                # Early termination check
-                if not (v_f > v_h).reduce_or():
-                    break_out = True
-                    break
-            if break_out:
-                break
+        @parameter
+        @always_inline
+        fn lazy_f():
+            @parameter
+            for _k in range(0, width):
+                var tmp = (
+                    ZERO - gap_open_penalty
+                ) if free_target_start_gaps else (
+                    boundary.unsafe_get(i + 1) - gap_open_penalty
+                )
+                var tmp2 = MIN if tmp < MIN else tmp.cast[dt]()
+
+                v_f = v_f.shift_right[1]()
+                v_f[0] = tmp2
+
+                for j in range(0, segment_length):
+                    v_h = pv_h_store[j]
+                    v_h = max(v_h, v_f)
+
+                    pv_h_store[j] = v_h
+
+                    @parameter
+                    if do_saturation_check:
+                        v_saturation_check_max = max(
+                            v_saturation_check_max, v_h
+                        )
+                        v_saturation_check_min = max(
+                            v_saturation_check_min, v_h
+                        )
+
+                    v_h = saturating_sub(v_h, v_gap_open)
+                    v_f = saturating_sub(v_f, v_gap_ext)
+
+                    # Early termination check - match C version exactly
+                    if not (v_f > v_h).reduce_or():
+                        return
```

Comment:
An inline closure allows for something more like a `goto` than the kind of complex set of breaks we had before.